open_instruct/grpo_fast.py

```diff
@@ -951,12 +976,34 @@ def train(
                             kl_loss_stats[i] = kl3_stats[i] * args.beta
                         elif args.kl_estimator == "kl4":
                             kl_loss_stats[i] = kl4_stats[i] * args.beta
-                        pg_clipfrac_stats[i] = masked_mean(
-                            (pg_losses2 > pg_losses).float(), mb_response_masks_bool, args.masked_mean_axis
-                        )
-                        pg_loss_stats[i] = masked_mean(pg_loss_max, mb_response_masks_bool, args.masked_mean_axis)
-                        loss_stats[i] = loss
-                        ratio_stats[i] = masked_mean(ratio, mb_response_masks_bool, args.masked_mean_axis)
+                        # pg_clipfrac_stats[i] = masked_mean(
+                        #     (pg_losses2 > pg_losses).float(), mb_response_masks_bool, args.masked_mean_axis
+                        # )
+                        # pg_loss_stats[i] = masked_mean(pg_loss_max, mb_response_masks_bool, args.masked_mean_axis)
+                        # loss_stats[i] = loss if not args.aggregate_maj else 0.0
+                        # ratio_stats[i] = masked_mean(ratio, mb_response_masks_bool, args.masked_mean_axis)
+                if args.aggregate_maj:    
+                    logprobs_diffs = torch.stack(logprob_diffs).reshape(-1, args.num_samples_per_prompt_rollout)
+                    
+
+                    sum_ratios = torch.exp(torch.sum(logprobs_diffs, dim=-1)).unsqueeze(-1)
+                    single_advantages = torch.stack(single_advantages).reshape(-1, args.num_samples_per_prompt_rollout)
```

Comment:
I worry a bit about the stacking. Maybe pre allocate and fill, try torch.cat, or removing some of the stacks?